modules/bwa/sampe/meta.yml

```diff
@@ -0,0 +1,58 @@
+name: bwa_sampe
+description: Convert paired-end bwa SA coordinate files to SAM format
+keywords:
+  - bwa
+  - aln
+  - short-read
+  - align
+  - reference
+  - fasta
+  - map
+  - sam
+  - bam
+tools:
+    - bwa:
+        description: |
+            BWA is a software package for mapping DNA sequences against
+            a large reference genome, such as the human genome.
+        homepage: http://bio-bwa.sourceforge.net/
+        documentation: http://bio-bwa.sourceforge.net/
+        doi: "10.1093/bioinformatics/btp324"
+      licence: ['GPL v3']
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - reads:
+      type: file
+      description: FASTQ files
+      pattern: "*.{fastq,fq}.gz"
+  - sai:
+      type: file
+      description: SAI file
+      pattern: "*.sai"
+  - index:
+        type: file
+        description: BWA genome index files
+        pattern: "Directory containing BWA index *.{amb,ann,bwt,pac,sa}"
```

Comment:
I understand that it should be a directory containing those 5 files. 
However `pattern` should be a valid glob pattern, and the given pattern matches *any* of those files, not all of them. 

With `STAR` we just used `star` as pattern: 
https://github.com/nf-core/modules/blob/2a7c60e9652fc37ad39e784f1b87b218fae8c94c/modules/star/align/meta.yml#L30

Not sure what's best.

Reply:
Oops, that was a copy-paste error for the pattern.

But re the variable: I also thought that, but I had to follow what was done in [`BOWTIE2`](https://github.com/nf-core/modules/blob/2a7c60e9652fc37ad39e784f1b87b218fae8c94c/modules/bowtie2/align/main.nf#L37-L48) as reference (which works).

When I just supply `$index` it gets an emitted variable of `./bwa/genome` (which is the directory and reference prefix), I'm not sure why tbh. The `$INDEX` gubbins is to get just the reference directory i.e. `bwa/` from the `bwa index` module that `bwa sampe` accepts.

I've updated the pattern (somewhat) acordingly.